### PR TITLE
Update NuGet and Docker dependencies

### PR DESCRIPTION
**Pull request by joperezr**
## Description

Updating dependencies from NuGet and Docker.

## Checklist

- Is this feature complete?
  - [x] Yes. Ready to ship.
  - [ ] No. Follow-up changes expected.
- Are you including unit tests for the changes and scenario tests if relevant?
  - [ ] Yes
  - [x] No
- Did you add public API?
  - [ ] Yes
    - If yes, did you have an API Review for it?
      - [ ] Yes
      - [ ] No
    - Did you add `<remarks />` and `<code />` elements on your triple slash comments?
      - [ ] Yes
      - [ ] No
  - [x] No
- Does the change make any security assumptions or guarantees?
  - [ ] Yes
    - If yes, have you done a threat model and had a security review?
      - [ ] Yes
      - [ ] No
  - [x] No
- Does the change require an update in our Aspire docs?
  - [ ] Yes
    - Link to aspire-docs issue: 
  - [x] No

 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/dotnet/aspire/pull/5575)

### CODE REVIEW

File: Directory.Packages.props [hunk 1]
```diff
@@ -93,19 +93,19 @@
...
     <PackageVersion Include="Grpc.AspNetCore" Version="2.65.0" />
     <PackageVersion Include="Grpc.Net.ClientFactory" Version="2.65.0" />
-    <PackageVersion Include="Grpc.Tools" Version="2.65.0" />
+    <PackageVersion Include="Grpc.Tools" Version="2.66.0" />
     <PackageVersion Include="Humanizer.Core" Version="2.14.1" />
     <PackageVersion Include="KubernetesClient" Version="14.0.9" />
     <PackageVersion Include="Microsoft.Data.SqlClient" Version="5.2.2" />
-    <PackageVersion Include="Microsoft.FluentUI.AspNetCore.Components" Version="4.9.3" />
-    <PackageVersion Include="Microsoft.FluentUI.AspNetCore.Components.Icons" Version="4.9.3" />
+    <PackageVersion Include="Microsoft.FluentUI.AspNetCore.Components" Version="4.10.0" />
```
Comment (sebastienros) [inline Directory.Packages.props:107]: Ping some devs from the dashboard to check we want this? UI can be picky.

File: src/Aspire.Hosting.Elasticsearch/ElasticsearchContainerImageTags.cs [hunk 2]
```diff
@@ -7,6 +7,6 @@ internal static class ElasticsearchContainerImageTags
 {
     public const string Registry = "docker.io";
     public const string Image = "library/elasticsearch";
-    public const string Tag = "8.15.0";
+    public const string Tag = "8.15.1";
```
Comment (sebastienros) [inline src/Aspire.Hosting.Elasticsearch/ElasticsearchContainerImageTags.cs:10]: @eerhardt @radical we need to update the feed, right? I can do it if you say yes

File: Directory.Packages.props (hunk 1 above)
Comment (sebastienros) [reply Directory.Packages.props:107]: @adamint @JamesNK can you confirm this is OK?

File: src/Aspire.Hosting.Elasticsearch/ElasticsearchContainerImageTags.cs (hunk 2 above)
Comment (joperezr) [reply src/Aspire.Hosting.Elasticsearch/ElasticsearchContainerImageTags.cs:10]: Correct, we need to update the registry. I was going to take care of it but if you're planning on doing it then that works for me too.

**Review by eerhardt — APPROVED** [review]
LGTM. Would be good to wait for the dashboard to confirm the new fluent UI versions.

File: Directory.Packages.props (hunk 1 above)
Comment (JamesNK) [reply Directory.Packages.props:107]: It is ok or not based on whether anything in the dashboard is broken. The only way to check is go through the dashboard UI and check to see if anything is broken.

Comment (joperezr) [reply Directory.Packages.props:107]: What am I looking for? Would it be better to just revert this and have someone in the dashboard side to update and validate instead?

Comment (JamesNK) [reply Directory.Packages.props]: You look at everything. But I created a PR here for it: https://github.com/dotnet/aspire/pull/5670

There are regressions.

Comment (joperezr) [reply Directory.Packages.props]: Yup, two tests were failing, I've reverted those changes from this PR to unblock it.

**Comment by eerhardt** [comment]
@joperezr - were you planning on backporting this change to release/8.2?

**Pull request merged**